src/chunk_append/exec.c

```diff
@@ -442,8 +444,25 @@ choose_next_subplan_for_worker(ChunkAppendState *state)
 {
 	LWLockAcquire(state->lock, LW_EXCLUSIVE);
 
-	state->current = get_next_subplan(state, state->pstate->last_plan);
-	state->pstate->last_plan = state->current;
+	if (state->current < state->pstate->last_plan &&
+		state->pstate->workers_last_plan < state->pstate->workers_per_child)
+	{
+		/*
+		 * if we have more workers then children we will assign
+		 * multiple workers per child
+		 */
+		Assert(state->pstate->workers_per_child > 1);
+		Assert(state->pstate->workers_last_plan >= 1);
+
+		state->current = state->pstate->last_plan;
```

Comment:
we will assign all remaining workers to the last plan? This looks wrong? Shouldn't we reassign to the subplan with fewest workers?

Reply:
we dont, we assign at most workers_per_child to a child